packages/core/src/linter/conditionSchema.ts

```diff
@@ -44,16 +44,53 @@ export function getConditionsJoiSchema(
         "notIn",
       )
       .required(),
-    value: Joi.alternatives()
-      .try(
-        // @TODO: make them more specific
-        Joi.string(),
-        Joi.number(),
-        Joi.boolean(),
-        Joi.date(),
-        Joi.array().items(Joi.string()),
-      )
-      .required(),
+    value: Joi.when('operator', {
+      is: Joi.string().valid(
+        "equals",
+        "notEquals"
+      ),
+      then: Joi.alternatives()
+        .try(
+          Joi.string(),
+          Joi.number(),
+          Joi.boolean(),
+          Joi.date(),
+        ).allow(null).required(),
+      otherwise: Joi.when("operator", {
+        is: Joi.string().valid(
+          "greaterThan",
+          "greaterThanOrEquals",
+          "lessThan",
+          "lessThanOrEquals"
+        ),
+        then: Joi.number().required(),
+        otherwise: Joi.when("operator", {
+          is: Joi.string().valid(
+            "contains",
+            "notContains",
+            "startsWith",
+            "endsWith",
+          ),
+          then: Joi.string().required(),
+          otherwise: Joi.when("operator", {
+            is: Joi.string().valid(
+              "semverEquals",
+              "semverNotEquals",
+              "semverGreaterThan",
+              "semverGreaterThanOrEquals",
+              "semverLessThan",
+              "semverLessThanOrEquals"
+            ),
+            then: Joi.string().required(),
+            otherwise: Joi.when("operator", {
+              is: Joi.string().valid("before", "after"),
+              then: Joi.date().required(),
```

Comment:
it is possible that dates can also be specified as string like "2023-10-12" without any timezone info.

in those cases, we have to accept it as `Joi.string()`.